rust/lance/src/index.rs

```diff
@@ -1461,6 +1468,136 @@ impl DatasetIndexInternalExt for Dataset {
             })
             .collect()
     }
+
+    async fn initialize_index(&mut self, source_dataset: &Dataset, index_name: &str) -> Result<()> {
+        // Load indices with the specified name from the source dataset
+        let source_indices = source_dataset.load_indices_by_name(index_name).await?;
+
+        if source_indices.is_empty() {
+            return Err(Error::Index {
+                message: format!("Index '{}' not found in source dataset", index_name),
+                location: location!(),
+            });
+        }
+
+        // Find the oldest (original) index by created_at timestamp, not delta indices
+        let source_index = source_indices
+            .iter()
+            .min_by_key(|idx| idx.created_at)
+            .ok_or_else(|| Error::Index {
+                message: format!("Could not determine oldest index for '{}'", index_name),
+                location: location!(),
+            })?;
+
+        // Check that ALL fields from the source index exist in the target dataset
+        let mut field_names = Vec::new();
+        for field_id in source_index.fields.iter() {
+            let source_field = source_dataset
+                .schema()
+                .field_by_id(*field_id)
+                .ok_or_else(|| Error::Index {
+                    message: format!("Field with id {} not found in source dataset", field_id),
+                    location: location!(),
+                })?;
+
+            // Check if the field exists in the current dataset
+            let target_field =
+                self.schema()
+                    .field(&source_field.name)
+                    .ok_or_else(|| Error::Index {
+                        message: format!(
+                            "Field '{}' required by index '{}' not found in target dataset",
+                            source_field.name, index_name
+                        ),
+                        location: location!(),
+                    })?;
+
+            // Verify field types match
+            if source_field.data_type() != target_field.data_type() {
+                return Err(Error::Index {
+                    message: format!(
+                        "Field '{}' has different types in source ({:?}) and target ({:?}) datasets",
+                        source_field.name,
+                        source_field.data_type(),
+                        target_field.data_type()
+                    ),
+                    location: location!(),
+                });
+            }
+
+            field_names.push(source_field.name.as_str().to_string());
+        }
+
+        if field_names.is_empty() {
+            return Err(Error::Index {
+                message: format!("Index '{}' has no fields", index_name),
+                location: location!(),
+            });
+        }
+
+        let column_name = field_names[0].as_str();
```

Comment:
this is based on the assumption of no multicolumn indexes I think - that works for now but we'll need to be careful not to break it when those are introduced.

Reply:
probably we should either pass a vec all the way through if we can (I know some of our methods work on vecs already for this reason), or error here if there's more than one column